src/scripts/main.js

```diff
@@ -1,3 +1,43 @@
 'use strict';
 
-// write code here
+function salaryToNumber(salary) {
+  return +salary
+    .split('')
+    .filter((char) => '1234567890'.includes(char))
+    .join('');
+}
+
+function sortList(list) {
+  const sortEmployees = list;
+
+  sortEmployees.sort(
+    (emp1, emp2) => salaryToNumber(emp2.salary) - salaryToNumber(emp1.salary),
+  );
+
+  return sortEmployees;
+}
+
+function getEmployees(listElement) {
+  const listOfEmployees = [];
+
+  for (const element of listElement) {
+    listOfEmployees.push({
+      name: element.textContent,
+      position: element.getAttribute('data-position'),
+      salary: element.getAttribute('data-salary'),
+      age: element.getAttribute('data-age'),
+    });
+  }
+
+  return listOfEmployees;
+}
+
+const listOfElements = document.querySelectorAll('li');
+const sortedEmployees = sortList(getEmployees(listOfElements));
+
+for (let i = 0; i < listOfElements.length; i++) {
+  listOfElements[i].innerHTML = sortedEmployees[i].name;
```

Comment:
Using `innerHTML` to set the text content can lead to security vulnerabilities such as XSS (Cross-Site Scripting) if the data is not properly sanitized. Consider using `textContent` instead to avoid these issues.